### DISCUSSION TITLE
Questions about MuWire

### DISCUSSION THREAD

**Discussion by ghost** [zlatinb/muwire#177] [Q&A]
Why do you think MuWire never became popular? Even though MuWire is the best designed p2p program that I know of I didn't find anything or anyone when I tried it and so I had to end up using older programs that had more people like aMule, EiskaltDC++ and gtk-gnutella.

Did you accomplish the idea you had in mind of the program when you started or did you still have some ideas planned for the future when you abandoned it?

Since this project interests me, I may want to maintain it, but I don't know the first thing about network protocols. What resources would you recommend to be able to understand this project.

One of the things I would do is creating a double identity so that a user can have different files shared over I2P and clearnet since I think having to configure I2P is what keeps people away from this program. Would the clearnet protocol be pretty much the same as the one for I2P or would I have to make a new one?

**Comment by zlatinb** [comment]
You ask many interesting questions and most of them are really not of technical nature, and for some I don't even have the answer.

>Did you accomplish the idea you had in mind of the program when you started or did you still have some ideas planned for the future when you abandoned it?

Specifically for the goals, I can say that one of the MuWire goals was to create a blueprint of sorts for a file publishing network.  I think I have done a decent job of documenting the internals as well as what is required to maintain such a network.

>Since this project interests me, I may want to maintain it, but I don't know the first thing about network protocols. What resources would you recommend to be able to understand this project.

The best thing to do is to study the specifics of the I2P protocols and the Gnutella network, version 0.4.  I suggest to start looking at the `doc` folder which contains various documents and then take it from there.
